cloud/gke.go

```diff
@@ -0,0 +1,1038 @@
+		Zone:      cs.Location,
+	}
+
+	response, err := GetClusterGoogle(svc, gkec)
+	if err != nil {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Something went wrong under read:", err)
+		return nil
+	} else {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Read cluster success")
+		clust := ClusterRepresentation{
+			Id:        cs.ID,
+			Name:      cs.Name,
+			CloudType: banzaiConstants.Google,
+			GoogleRepresentation: &GoogleRepresentation{
+				GoogleCluster: response,
+			},
+		}
+		return &clust
+	}
+}
+
+type GoogleRepresentation struct {
+	GoogleCluster *gke.Cluster `json:"value,omitempty"`
+}
+
+func GetClusterInfoGoogle(cs *banzaiSimpleTypes.ClusterSimple, c *gin.Context) {
+	banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Fetch aks cluster with name:", cs.Name, "in", cs.Azure.ResourceGroup, "resource group.")
+
+	if cs == nil {
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "<nil> cluster")
+		SetResponseBodyJson(c, http.StatusInternalServerError, gin.H{
+			JsonKeyStatus: http.StatusInternalServerError,
+		})
+		return
+	}
+
+	svc, err := GetGoogleServiceClient()
+	if err != nil {
+		SetResponseBodyJson(c, http.StatusInternalServerError, gin.H{
+			JsonKeyStatus:  http.StatusInternalServerError,
+			JsonKeyMessage: err,
+		})
+		return
+	}
+
+	gkec := GKECluster{
+		ProjectID: cs.Google.Project,
+		Name:      cs.Name,
+		Zone:      cs.Location,
+	}
+
+	response, err := GetClusterGoogle(svc, gkec)
+	if err != nil {
+		// fetch failed
+		googleApiErr := getBanzaiErrorFromError(err)
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Status code:", googleApiErr.StatusCode)
+		banzaiUtils.LogInfo(banzaiConstants.TagGetCluster, "Error during get cluster details:", googleApiErr.Message)
+		SetResponseBodyJson(c, googleApiErr.StatusCode, googleApiErr)
+	} else {
+		// fetch success
+		SetResponseBodyJson(c, http.StatusOK, gin.H{
+			JsonKeyResourceId: cs.ID,
+			JsonKeyData:       response,
+		})
+	}
+
+}
+
+// UpdateClusterGoogleInCloud updates google cluster in cloud
+func UpdateClusterGoogleInCloud(r *banzaiTypes.UpdateClusterRequest, c *gin.Context, preCluster banzaiSimpleTypes.ClusterSimple) bool {
+
+	banzaiUtils.LogInfo(banzaiConstants.TagUpdateCluster, "Start updating cluster (google)")
+
+	if r == nil {
+		banzaiUtils.LogInfo(banzaiConstants.TagUpdateCluster, "<nil> update cluster")
+		return false
+	}
+
+	cluster2Db := banzaiSimpleTypes.ClusterSimple{
+		Model:            preCluster.Model,
+		Name:             preCluster.Name,
+		Location:         preCluster.Location,
+		NodeInstanceType: preCluster.NodeInstanceType,
+		Cloud:            preCluster.Cloud,
+		Google: banzaiSimpleTypes.GoogleClusterSimple{
+			Project:       preCluster.Google.Project,
+			NodeCount:     r.GoogleNode.Count,
+			MasterVersion: r.GoogleMaster.Version,
+			NodeVersion:   r.GoogleNode.Version,
+		},
+	}
+
+	svc, err := GetGoogleServiceClient()
+	if err != nil {
+		SetResponseBodyJson(c, http.StatusInternalServerError, gin.H{
```

Comment:
Log the error.

Reply:
If there was an error, the GetGoogleServiceClient method will log